config/env_config.go

```diff
@@ -335,6 +335,11 @@ func (c EnvConfig) getDefaultsMode(ctx context.Context) (aws.DefaultsMode, bool,
 	return c.DefaultsMode, true, nil
 }
 
+func (c EnvConfig) getAppID(context.Context) (string, bool, error) {
+	appID := os.Getenv(`AWS_SDK_UA_APP_ID`)
```

Comment:
while this works, i dont think this is the typical setup. see the other fields in `env_config.go`

Reply:
Just want to lazy set this field in case it is modified, but you are right, I will change that

Reply:
i dont think there is actually any reasonable time interval between when `NewEnvConfig` is called and when `getAppId` is called. so i dont think lazily evaluating it here makes a difference

Reply:
Yeah - `loadEnvConfig` first reads everything out of the environment into the env config struct, and then the "resolvers" e.g. `getAppID` just read out of the struct. `getAppID` shouldn't be querying the environment directly. See other fields in `env_config` as @isaiahvita said.

Reply:
> Just want to lazy set this field in case it is modified,

> i dont think there is actually any reasonable time interval

There's no change of changing or interval to speak of - a process' environment is static.